ninja_simple_jwt/auth/backends.py

```diff
@@ -0,0 +1,100 @@
+from jwt import PyJWTError
+
+from ninja_simple_jwt.jwt.token_operations import TokenTypes, decode_token
+from ninja_simple_jwt.settings import ninja_simple_jwt_settings
+
+User = get_user_model()  # type: ignore[assignment]
+TokenUser = import_string(ninja_simple_jwt_settings.TOKEN_USER_CLS)
+
+
+class TokenUserBackend(BaseBackend):
+    def authenticate(
+        self, request: Optional[HttpRequest], token: Optional[str] = None, **kwargs: Any
+    ) -> Optional[AbstractBaseUser]:
+        """
+        Authenticate a stateless user using a JWT token.
+
+        Args:
+            request: The HTTP request object
+            token: The JWT token string
+
+        Returns:
+            User instance if authentication successful, None otherwise
+        """
+        if not token:
+            return None
+
+        try:
+            # Decode and verify the JWT token
+            decoded_token = decode_token(token, token_type=TokenTypes.ACCESS, verify=True)
+
+            # Extract user_id from the token
+            user_id = decoded_token.get("user_id")
+            if not user_id:
+                return None
+
+            # Create a stateless user instance from token claims
+            user = self._create_user_from_token(decoded_token)
+            return user
+
+        except PyJWTError:
+            # Token is invalid, expired, or malformed
+            return None
+
+    @staticmethod
+    def _create_user_from_token(token: dict) -> AbstractBaseUser:
+        """
+        Create a user instance from JWT token claims without database access.
+
+        This creates a "fake" User instance that Django ORM will accept for queries
+        by properly setting its primary key and state.
+
+        Args:
+            token: The decoded JWT token payload
+
+        Returns:
+            User instance populated with token claims
+        """
+
+        user = TokenUser(user_id=token.get("user_id"), email=token.get("email", None), roles=token.get("roles", []))
+
+        # Set the primary key attribute on the TokenUser object
+        setattr(user, User._meta.pk.name, token.get("user_id"))
+
+        # Set all token claims as attributes on the user object
+        for claim in ninja_simple_jwt_settings.TOKEN_CLAIM_USER_ATTRIBUTE_MAP.keys():
+            if claim in token:
+                setattr(user, claim, token.get(claim))
+
+        return user
+
+
+class ModelUserBackend(BaseBackend):
+    def authenticate(
+        self, request: Optional[HttpRequest], token: Optional[str] = None, **kwargs: Any
+    ) -> Optional[AbstractBaseUser]:
+        """
+        Authenticate a token using database User model.
+        """
+        if not token:
+            return None
+
+        try:
+            decoded_token = decode_token(token, token_type=TokenTypes.ACCESS, verify=True)
+
+            user_id = decoded_token.get("user_id")
+            if not user_id:
+                return None
+
+            user = User.objects.get(id=user_id)
```

Comment:
I think you might want to go with a `.first()` here instead so that you get `None` instead of raising an error.

```suggestion
            user = User.objects.filter(id=user_id).first()
```